linera-base/src/crypto/signer.rs

```diff
@@ -0,0 +1,315 @@
+        ) -> Result<AccountSignature, Box<dyn std::error::Error>> {
+            let inner = self.0.read().unwrap();
+            if let Some(secret) = inner.keys.get(owner) {
+                let signature = secret.sign_prehash(*value);
+                Ok(signature)
+            } else {
+                Err("No key found for the given owner".into())
+            }
+        }
+
+        /// Returns the public key corresponding to the given `owner`.
+        async fn get_public_key(
+            &self,
+            owner: &AccountOwner,
+        ) -> Result<AccountPublicKey, Box<dyn std::error::Error>> {
+            let inner = self.0.read().unwrap();
+            match inner.keys.get(owner).map(|s| s.public()) {
+                Some(public) => Ok(public),
+                None => Err("No key found for the given owner".into()),
+            }
+        }
+
+        /// Returnes whether the given `owner` is a known signer.
+        async fn contains_key(
+            &self,
+            owner: &AccountOwner,
+        ) -> Result<bool, Box<dyn std::error::Error>> {
+            let inner = self.0.read().unwrap();
+            Ok(inner.keys.contains_key(owner))
+        }
+    }
+
+    impl FromIterator<(AccountOwner, AccountSecretKey)> for InMemorySigner {
+        fn from_iter<T>(input: T) -> Self
+        where
+            T: IntoIterator<Item = (AccountOwner, AccountSecretKey)>,
+        {
+            InMemorySigner(Arc::new(RwLock::new(InMemSignerInner {
+                keys: BTreeMap::from_iter(input),
+                #[cfg(with_getrandom)]
+                rng_state: RngState::new(None),
+            })))
+        }
+    }
+
+    impl Default for InMemSignerInner {
+        fn default() -> Self {
+            #[cfg(with_getrandom)]
+            let signer = InMemSignerInner::new(None);
+            #[cfg(not(with_getrandom))]
+            let signer = InMemSignerInner::new();
+            signer
+        }
+    }
+
+    impl Serialize for InMemorySigner {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: serde::Serializer,
+        {
+            let inner = self.0.read().unwrap();
+            InMemSignerInner::serialize(&*inner, serializer)
+        }
+    }
+
+    impl<'de> Deserialize<'de> for InMemorySigner {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: serde::Deserializer<'de>,
+        {
+            let inner = InMemSignerInner::deserialize(deserializer)?;
+            Ok(InMemorySigner(Arc::new(RwLock::new(inner))))
+        }
+    }
+
+    impl Serialize for InMemSignerInner {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: serde::Serializer,
+        {
+            #[derive(Serialize, Debug)]
+            struct Inner<'a> {
+                keys: &'a Vec<(AccountOwner, Vec<u8>)>,
+                #[cfg(with_getrandom)]
+                prng_seed: Option<u64>,
```

Comment:
I think since we're not trying to allow recreation of seed states we don't need to store this one, and can just generate a new seed from the RNG when we save :thinking: